File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n = matrix.length;
        int m = matrix[0].length;
        int j = 0;
        int l = n - 1;
        while (j < m && l > -1) {
            int s = 0;
            while (s <= l) {
                int mid = (s + l) / 2;
                if (matrix[mid][j] == target) return true;
                if (matrix[mid][j] < target) s = mid + 1;
                else l = mid - 1;
            }
            j++;
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n, m = len(matrix), len(matrix[0])
        l = n - 1
        i = 0
        while l > -1 and i < m:
            s = 0
            while s <= l:
                mid = s + (l - s) // 2
                if matrix[mid][i] == target:
                    return True
                if matrix[mid][i] > target:
                    l = mid - 1
                else:
                    s = mid + 1
            i += 1
        return False